### PR TITLE
MLPAB-1055: Start to document data for wider team visibility

### PR DESCRIPTION
**Pull request by acsauk**
# Purpose

This is a starting point to get some feedback from the wider team on the format and level of detail here. Once we have a better feel for team members' needs we can iterate on this to get to where we need to be.

As we've updated the module name to show the docs on pkg.go.dev there's a bunch of noise in the PR changing imports so I've flagged files with actual changes for review.

Fixes MLPAB-1055

### CODE REVIEW

File: app/internal/actor/attorney.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 	"golang.org/x/exp/slices"
 )
 
+// Attorney contains details about an attorney or replacement attorney, provided by the applicant
 type Attorney struct {
```
Comment (acsauk) [inline app/internal/actor/attorney.go:13]: Comments added here

File: app/internal/actor/attorney_decisions.go [hunk 2]
```diff
@@ -9,7 +9,7 @@ const (
 	JointlyForSomeSeverallyForOthers = "mixed"
 )
 
-// AttorneyDecisions collects how a donor wants their attorneys or replacement attorneys to act
+// AttorneyDecisions contains details about how an attorney or replacement attorney should act, provided by the applicant
 type AttorneyDecisions struct {
```
Comment (acsauk) [inline app/internal/actor/attorney_decisions.go:13]: Comments added here

File: app/internal/actor/certificate_provider.go [hunk 3]
```diff
@@ -6,6 +6,7 @@ import (
 	"github.com/ministryofjustice/opg-modernising-lpa/internal/place"
 )
 
+// CertificateProvider contains details about the certificate provider, provided by the applicant
 type CertificateProvider struct {
```
Comment (acsauk) [inline app/internal/actor/certificate_provider.go:10]: Comments added here

File: app/internal/actor/certificate_provider_provided.go [hunk 4]
```diff
@@ -9,7 +9,7 @@ import (
 	"github.com/ministryofjustice/opg-modernising-lpa/internal/place"
 )
 
-// CertificateProviderProvidedDetails are details provided by the certificate provider
+// CertificateProviderProvidedDetails contains details about the certificate provider, provided by the certificate provider
 type CertificateProviderProvidedDetails struct {
```
Comment (acsauk) [inline app/internal/actor/certificate_provider_provided.go:13]: Comments added here

File: app/internal/actor/donor.go [hunk 5]
```diff
@@ -7,6 +7,7 @@ import (
 	"github.com/ministryofjustice/opg-modernising-lpa/internal/place"
 )
 
+// Donor contains details about the donor, provided by the applicant
 type Donor struct {
```
Comment (acsauk) [inline app/internal/actor/donor.go:11]: Comments added here

File: app/internal/actor/person_to_notify.go [hunk 6]
```diff
@@ -5,6 +5,7 @@ import (
 	"golang.org/x/exp/slices"
 )
 
+// PersonToNotify contains details about a person to notify, provided by the applicant
 type PersonToNotify struct {
```
Comment (acsauk) [inline app/internal/actor/person_to_notify.go:9]: Comments added here

File: app/internal/page/data.go [hunk 7]
```diff
@@ -1,14 +1,14 @@
...
-//   - [actor.PersonToNotify] - a struct containing details about a person to notify, provided by the applicant
+//   - [page.Lpa] - details about the LPA being drafted
+//   - [actor.Donor] - details about the donor, provided by the applicant
+//   - [actor.CertificateProvider] - details about the certificate provider, provided by the applicant
+//   - [actor.CertificateProviderProvidedDetails] - details about the certificate provider, provided by the certificate provider
+//   - [actor.Attorney] - details about an attorney or replacement attorney, provided by the applicant
+//   - [actor.AttorneyDecisions] - details about how an attorney or replacement attorney should act, provided by the applicant
+//   - [actor.AttorneyProvidedDetails] - details about an attorney or replacement attorney, provided by the attorney or replacement attorney
+//   - [actor.PersonToNotify] - details about a person to notify, provided by the applicant
 package page
```
Comment (acsauk) [inline app/internal/page/data.go:12]: Comments added here

File: app/internal/page/data.go [hunk 8]
```diff
@@ -36,7 +36,7 @@ const (
 	OptionB                = "option-b"
 )
 
-// Lpa is all the data related to the LPA application
+// Lpa contains all the data related to the LPA application
 type Lpa struct {
```
Comment (acsauk) [inline app/internal/page/data.go:40]: Comments added here

File: app/internal/actor/donor.go [hunk 9]
```diff
@@ -3,19 +3,27 @@ package actor
...
 	"fmt"
 
-	"github.com/ministryofjustice/opg-modernising-lpa/internal/date"
-	"github.com/ministryofjustice/opg-modernising-lpa/internal/place"
+	"github.com/ministryofjustice/opg-modernising-lpa/app/internal/date"
+	"github.com/ministryofjustice/opg-modernising-lpa/app/internal/place"
 )
 
+// Donor contains details about the donor, provided by the applicant
 type Donor struct {
```
Comment (acsauk) [inline app/internal/actor/donor.go:11]: Comments added here

File: app/internal/actor/person_to_notify.go [hunk 10]
```diff
@@ -1,16 +1,22 @@
 package actor
 
 import (
-	"github.com/ministryofjustice/opg-modernising-lpa/internal/place"
+	"github.com/ministryofjustice/opg-modernising-lpa/app/internal/place"
 	"golang.org/x/exp/slices"
 )
 
+// PersonToNotify contains details about a person to notify, provided by the applicant
 type PersonToNotify struct {
```
Comment (acsauk) [inline app/internal/actor/person_to_notify.go:9]: Comments added here

File: app/internal/actor/attorney.go [hunk 11]
```diff
@@ -4,18 +4,25 @@ import (
...
 
-	"github.com/ministryofjustice/opg-modernising-lpa/internal/date"
-	"github.com/ministryofjustice/opg-modernising-lpa/internal/place"
+	"github.com/ministryofjustice/opg-modernising-lpa/app/internal/date"
+	"github.com/ministryofjustice/opg-modernising-lpa/app/internal/place"
 	"golang.org/x/exp/slices"
 )
 
+// Attorney contains details about an attorney or replacement attorney, provided by the applicant
 type Attorney struct {
```
Comment (acsauk) [inline app/internal/actor/attorney.go:13]: Comments added here

File: app/internal/actor/attorney_decisions.go [hunk 12]
```diff
@@ -1,15 +1,23 @@
...
+	// Jointly indicates attorneys or replacement attorneys should act jointly
+	Jointly = "jointly"
+	// JointlyAndSeverally attorneys or replacement attorneys should act jointly and severally
+	JointlyAndSeverally = "jointly-and-severally"
+	// JointlyForSomeSeverallyForOthers attorneys or replacement attorneys should act jointly for some decisions, and jointly and severally for other decisions
 	JointlyForSomeSeverallyForOthers = "mixed"
 )
 
+// AttorneyDecisions contains details about how an attorney or replacement attorney should act, provided by the applicant
 type AttorneyDecisions struct {
```
Comment (acsauk) [inline app/internal/actor/attorney_decisions.go:13]: Comments added here

File: app/internal/actor/attorney_provided.go [hunk 13]
```diff
@@ -3,23 +3,34 @@ package actor
...
 	"time"
 
-	"github.com/ministryofjustice/opg-modernising-lpa/internal/date"
-	"github.com/ministryofjustice/opg-modernising-lpa/internal/place"
+	"github.com/ministryofjustice/opg-modernising-lpa/app/internal/date"
+	"github.com/ministryofjustice/opg-modernising-lpa/app/internal/place"
 )
 
+// AttorneyProvidedDetails contains details about an attorney or replacement attorney, provided by the attorney or replacement attorney
 type AttorneyProvidedDetails struct {
```
Comment (acsauk) [inline app/internal/actor/attorney_provided.go:11]: Comments added here

File: app/internal/page/data.go [hunk 14]
```diff
@@ -25,43 +36,77 @@ const (
 	OptionB                = "option-b"
 )
 
+// Lpa contains all the data related to the LPA application
 type Lpa struct {
```
Comment (acsauk) [inline app/internal/page/data.go:40]: Comments added here

**Review by acsauk — COMMENTED** [review]
I've highlighted the comments added to break through the module name updates noise

File: app/internal/actor/attorney.go [hunk 15]
```diff
@@ -4,18 +4,25 @@ import (
...
+	FirstNames string
+	// Last name of the attorney
+	LastName string
+	// Email of the attorney
+	Email string
+	// Date of birth of the attorney
 	DateOfBirth date.Date
-	Address     place.Address
+	// Address of the attorney
+	Address place.Address
```
Comment (hawx) [inline app/internal/actor/attorney.go:25]: I think all of these fields are self-explanatory so don't require comments

File: app/internal/actor/attorney.go [hunk 16]
```diff
@@ -4,18 +4,25 @@ import (
...
 	"strings"
 
-	"github.com/ministryofjustice/opg-modernising-lpa/internal/date"
-	"github.com/ministryofjustice/opg-modernising-lpa/internal/place"
+	"github.com/ministryofjustice/opg-modernising-lpa/app/internal/date"
+	"github.com/ministryofjustice/opg-modernising-lpa/app/internal/place"
 	"golang.org/x/exp/slices"
 )
 
+// Attorney contains details about an attorney or replacement attorney, provided by the applicant
```
Comment (hawx) [inline app/internal/actor/attorney.go:12]: this is a small thing, but generally structs would start with "A" or "An" and the fields/methods would start with the field or method name, e.g. https://pkg.go.dev/net/http#Client

**Review by hawx — APPROVED** [review]
I'd personally prefer to not comment where possible, but am happy for this to go in for now to see how it looks/if it will be useful to people, and iterate as we go

**Pull request merged**